kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/admin/ProcessAdminResource.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.remote.rest.jbpm.admin;
+
+import java.text.MessageFormat;
+import java.util.List;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Variant;
+
+import org.jbpm.services.api.DeploymentNotFoundException;
+import org.jbpm.services.api.ProcessInstanceNotFoundException;
+import org.kie.server.api.model.admin.MigrationReportInstance;
+import org.kie.server.api.model.admin.MigrationReportInstanceList;
+import org.kie.server.remote.rest.common.Header;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.jbpm.admin.ProcessAdminServiceBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.kie.server.api.rest.RestURI.*;
+import static org.kie.server.remote.rest.common.util.RestUtils.*;
+import static org.kie.server.remote.rest.jbpm.resources.Messages.*;
+
+@Path("server/" + ADMIN_PROCESS_URI)
+public class ProcessAdminResource {
+
+    public static final Logger logger = LoggerFactory.getLogger(ProcessAdminResource.class);
+
+    private ProcessAdminServiceBase processAdminServiceBase;
+    private KieServerRegistry context;
+
+
+    public ProcessAdminResource() {
+
+    }
+
+    public ProcessAdminResource(ProcessAdminServiceBase processAdminServiceBase, KieServerRegistry context) {
+        this.processAdminServiceBase = processAdminServiceBase;
+        this.context = context;
+    }
+
+    @PUT
+    @Path(MIGRATE_PROCESS_INST_PUT_URI)
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response migrateProcessInstance(@javax.ws.rs.core.Context HttpHeaders headers, @PathParam("id") String containerId, @PathParam("pInstanceId") Long processInstanceId,
```

Comment:
This method accepts parameter "pInstanceId" for process instance ID, the next one uses "instanceId" for a list of process instance IDs. Could we name it e.g. "instanceId" and "instanceIds" to clearly communicate cardinality of these parameters?